x-pack/plugins/fleet/server/routes/agent_policy/handlers.ts

```diff
@@ -109,37 +123,55 @@ export const createAgentPolicyHandler: FleetRequestHandler<
   const esClient = context.core.elasticsearch.client.asInternalUser;
   const user = (await appContextService.getSecurity()?.authc.getCurrentUser(request)) || undefined;
   const withSysMonitoring = request.query.sys_monitoring ?? false;
+  const { has_fleet_server: hasFleetServer, ...newPolicy } = request.body;
   const spaceId = context.fleet.spaceId;
   try {
-    // eslint-disable-next-line prefer-const
-    let [agentPolicy, newSysPackagePolicy] = await Promise.all([
-      agentPolicyService.create(soClient, esClient, request.body, {
+    if (hasFleetServer) {
```

Comment:
Let's move all of these `ensureInstalledPackage` calls to a single `bulkInstallPackages` call instead for slightly better performance.

Reply:
`bulkInstallPackages` doesn't help much though, one time it took 144 s (2.4 min) to install system + elastic_agent still.
it took 80s for elastic_agent, when system is already installed.

Previously I didn't add await for installing elastic agent (with await), so at least the create agent policy call could return that much earlier. Do you think that's a good idea?
The only caveat with that is, if someone enrolls an agent before elastic_agent package is fully installed, which seems unlikely.

Reply:
> Previously I didn't add await for installing elastic agent (with await), so at least the create agent policy call could return that much earlier. Do you think that's a good idea?
The only caveat with that is, if someone enrolls an agent before elastic_agent package is fully installed, which seems unlikely.

If the install fails for any reason, I think this could be a problem so I do think we need to `await` this before proceeding. We can use this as another reason to improve package install performance.